substrate/state-machine/src/lib.rs

```diff
@@ -183,12 +196,34 @@ pub fn execute<B: backend::Backend, Exec: CodeExecutor>(
 			.ok_or(Box::new(ExecutionError::CodeEntryDoesNotExist) as Box<Error>)?
 			.to_vec();
 
-		exec.call(
+		let (result, was_native) = exec.call(
 			&mut externalities,
 			&code,
 			method,
 			call_data,
-		).map(move |out| (out, externalities.transaction()))
+			strategy != ExecutionStrategy::AlwaysWasm,
+		);
+		let result = if was_native && strategy == ExecutionStrategy::Both {
+			let (wasm_result, _) = exec.call(
+				&mut externalities,
+				&code,
+				method,
+				call_data,
+				strategy != ExecutionStrategy::AlwaysWasm,
```

Comment:
This expression confuses me, why just don't simplify it to `true`? Why it is `true` in the first place?